homeassistant/components/caldav/calendar.py

```diff
@@ -199,6 +199,22 @@ def update(self):
             dt.start_of_local_day(), dt.start_of_local_day() + timedelta(days=1)
         )
 
+        # Change start/end dates of recurring events to today if they occur today.
+        # For recurring events, some servers return the original event with recurrence rules
+        # and they would not be properly parsed using their original start/end dates.
+        for event in results:
+            vevent = event.instance.vevent
+            rrules = vevent.getrruleset()
+            if rrules:
+                sld = dt.start_of_local_day()
+                if sld.date() in (rrule.date() for rrule in rrules):
```

Comment:
`rrules` can be indefinite?

Reply:
Also, shouldn't you try to find the first instance of an RRULE that is after now(). An event that recurs every 12 hours could have 2 instances today, one before now() and one after.

Reply:
Yes, `rrules` can return `None`.

Events cannot [repeat](https://icalendar.org/iCalendar-RFC-5545/3-8-5-3-recurrence-rule.html) more than daily. Intervals less than 24 hours would have to be two separate events.

Reply:
From your link:

Every 15 minutes for 6 occurrences:

```vcard
 DTSTART;TZID=America/New_York:19970902T090000
 RRULE:FREQ=MINUTELY;INTERVAL=15;COUNT=6
```

Also, RRuleSets can repeat forever. If it's a weekly one, current date might not be in it, causing the if check to hang forever.

Reply:
Wait - yes they can even be hourly or minutely, [apparently](https://tools.ietf.org/html/rfc5545). It's just that the three clients I tested won't create those rules.

Reply:
Or secondly!

Reply:
Correct. Most clients do not implement more frequent than `DAILY` because it can easily be used to DDOS their system (and it's seldom used).

Reply:
Well, I'm open to suggestions. :)

Reply:
Iterate over it until the first occurrence that is after now() or occurrences are exhausted.